core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1032,14 +1032,14 @@ private[spark] class TaskSetManager(
       return false
     }
     var foundTasks = false
-    val minFinishedForSpeculation = (speculationQuantile * numTasks).floor.toInt
+    val minFinishedForSpeculation = math.max((speculationQuantile * numTasks).floor.toInt, 1)
```

Comment:
sorry I had actually meant to just compute this once in the constructor, eg. around line 70